app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,43 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      scheme, instance_uri = provider_id.split("://", 2)
+      prov, name_field = {
+        "gce" => [ManageIQ::Providers::Google::CloudManager, :name],
+        "aws" => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }[scheme]
+      instance_id = instance_uri.split('/').last
+
+      prov::Vm.find_by(name_field => instance_id) if !prov.nil? && !instance_id.empty?
```

Comment:
@enoodle I had a reason to suggest `!instance_id.blank?` instead of `empty?`. And it's because it works with `nil` as well.